xclim/indices/generic.py

```diff
@@ -776,6 +780,59 @@ def degree_days(tas: xr.DataArray, thresh: str, condition: str) -> xr.DataArray:
         out = (thresh - tas).clip(0)
     elif ">" in condition:
         out = (tas - thresh).clip(0)
+    else:
+        raise NotImplementedError(f"Condition not supported: '{condition}'.")
 
     out = to_agg_units(out, tas, op="delta_prod")
     return out
+
+
+def day_length_coefficient(
+    dates: xr.DataArray,
+    lat: xr.DataArray,
+    obliquity: float = 0.4091,
+    start_date: Optional[Union[xarray.DataArray, DayOfYearStr]] = None,
+    end_date: Optional[Union[xarray.DataArray, DayOfYearStr]] = None,
+    freq: str = "YS",
+) -> xr.DataArray:
+    """Day-length coefficient.
+
+    Parameters
+    ----------
+    dates: xr.DataArray
+    lat: xarray.DataArray
+      Latitude coordinate.
+    obliquity: float
+      Obliquity of the elliptic (rads). Default: 0.4091.
+    start_date: xarray.DataArray or DayOfYearStr, optional
+    end_date: xarray.DataArray or DayOfYearStr, optional
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+      If start and end date provided, returns total sum of daylight-hour between dates at provided frequency.
+      If no start and end date provided, returns day-length in hours per individual day.
+
+    References
+    ----------
+    Modified day-length equations for Huglin heliothermal index published in Hall, A., & Jones, G. V. (2010). Spatial
+    analysis of climate in winegrape-growing regions in Australia. Australian Journal of Grape and Wine Research, 16(3),
+    389‑404. https://doi.org/10.1111/j.1755-0238.2010.00100.x
+
+    Examples available from Glarner, 2006 (http://www.gandraxa.com/length_of_day.xml).
+    """
+    warnings.warn("This is untested.", UserWarning, stacklevel=4)
```

Comment:
A simple test should be easy to add, no?

Reply:
True. This should be easy to do in `test_generic.py`.